### DISCUSSION TITLE
[Solved] Displaying chat in Player - should it work?

### DISCUSSION THREAD

**Discussion by mr-flibble** [MrBrax/LiveStreamDVR#380] [General]
Hello, should displaying chat in vod Player work?

Every time i try player with _captured_ chat, then chat is not visible and this error show in console after clicking "Start playback":
`index.cd3bd71f.js:5 Uncaught (in promise) TypeError: this.emotes is not iterable
    at Proxy.parseComment (index.cd3bd71f.js:5:11885)
    at Proxy.tick (index.cd3bd71f.js:5:34238)`

I tried disabling everything in "Chat style", but same error every time 

![image](https://user-images.githubusercontent.com/5197625/202795372-7a4824c9-f986-4094-849f-caa28fcdf2e4.png)

**Comment by MrBrax** [comment]
I really don't understand how this is happening anymore. I've tested it with multiple vods and it works for me. Is this a really weird edge case?

**Reply by mr-flibble** [reply]
I tested it with downloaded chat and same error. 

Maybe is problem with that channel, I can see this error in console:
`Failed to load resource: the server responded with a status of 404 ()`
for URL: https://api.7tv.app/v2/users/--workingID--/emotes

If I go to that url, this is content:
`{"status_code":404,"status":"Not Found","error":"No Items Found","error_code":70499}`


( I don't want to post real channel username/ ID  here - but if you need it, I could probably message it 

![image](https://user-images.githubusercontent.com/5197625/202800748-c28eb308-1541-49ed-a725-f38921647cfa.png)

**Comment by MrBrax** [comment]
ok i'm hoping the latest update fixes it, what a pain in the ass error

**Reply by mr-flibble** [reply]
Thanks!  Problem solved and chat works great! 
![image](https://user-images.githubusercontent.com/5197625/202806445-5da645bb-6596-439f-881b-04059979f012.png)


Is as intended that seeking in vod (clicking to another time) briefly displays all messages from the chat on the screen and when the video starts playing again chat disappear and new messages pop up gradually?

It seems like scrolling in the video causes "reset chat".

**Reply by MrBrax** [reply]
yes, seeking causes everything to reset because it has to check all messages for displayed status and where in the video it's in

**Comment by MrBrax** [comment]
also for the future, vod player is a different package: https://github.com/MrBrax/twitch-vod-chat

**Comment by mr-flibble** [comment]
Thanks and thanks !
